search/searcher/search_phrase.go

```diff
@@ -32,7 +32,7 @@ func init() {
 }
 
 type PhraseSearcher struct {
-	mustSearcher *ConjunctionSearcher
+	mustSearcher search.Searcher
```

Comment:
Why generalize this? It makes sense that the mustSearcher within a PhraseSearcher must always be a ConjunctionSearcher, yes?

Reply:
> Why generalize this? It makes sense that the mustSearcher within a PhraseSearcher must always be a ConjunctionSearcher, yes?

Indeed, but what's happening is the return type for the NewConjunctionSearcher() func was changed in this PR to return a search.Searcher (instead of a *ConjunctionSearcher).  So, users like the PhraseSearcher needed to change to search.Searcher as well.

The change in return type was motivated by the PR's optimization where sometimes NewConjunctionSearcher() might return a simple term searcher if the optimization kicks in.

This also makes NewConjunctionSearcher() look a little more symmetric with the NewDisjunctionSearcher() func, which was already returning a search.Searcher (instead of a *DisjunctionSearcher).